inst/@sym/sym.m

```diff
@@ -156,257 +156,312 @@
+      if (iscell (x))  % Handle Cells
+        error ('conversion to cell array of symbols from the constructor is no longer supported. Use sym.symarray instead.');
+        return
       end
-      x = strrep (x, '"', '\"');   % Avoid collision with S("x") and Symbol("x")
 
-      isnum = ~isempty (regexp (x, '^[-+]*?\d*\.?\d*(e-?\d+)?$'));  % Is Number
-    end
+      asm = {};
+      check = true;
 
-    %% Use Symbol() for words, not numbers, not "f(x)".
-    if (check && ~isnum && regexp (x, '^\w+$'))
-
-      cmd = { 'd = dict()'
-              '_ins = [_ins] if isinstance(_ins, dict) else _ins'
-              'for i in range(len(_ins)):'
-              '    if isinstance(_ins[i], dict):'
-              '        d.update(_ins[i])'
-              '    #elif isinstance(_ins[i], list):'  % TODO: allow a list?
-              '    #    for j in range(len(_ins[i])):'
-              '    #        d.update({_ins[i][j]:True})'
-              '    elif isinstance(_ins[i], (str, bytes)):'
-              '        d.update({_ins[i]:True})'
-              '    else:'
-              '        raise ValueError("something unexpected in assumptions")'
-              'return Symbol("{s}", **d)' };
-      s = python_cmd (strrep (cmd, '{s}', x), asm{:});
-
-      if (nargin == 2 && sclear)
-        % ---------------------------------------------
-        % Muck around in the caller's namespace, replacing syms
-        % that match 'xstr' (a string) with the 'newx' sym.
-        context = 'caller';
-        S = evalin(context, 'whos');
-        evalin(context, '[];');  % clear 'ans'
-        for i = 1:numel(S)
-          obj = evalin(context, S(i).name);
-          [newobj, flag] = symreplace(obj, x, s);
-          if flag, assignin(context, S(i).name, newobj); end
+      if (nargin >= 2)
+        if (ismatrix (varargin{1}) && ~isa (varargin{1}, 'char') && ~isstruct (varargin{1}) && ~iscell (varargin{1})) % Handle MatrixSymbols
+          error ('conversion to matrix of symbols from the constructor is no longer supported. Use sym.symarray instead.');
+          return
+        else
+          if (nargin == 2 && ischar(varargin{1}) && strcmp(varargin{1},'clear'))
+            sclear = true;
+            varargin(1) = [];
+            %warning ('deprecated: "sym(x, ''clear'')" will be removed in future version');
+          else
+            sclear = false;
+            sym.check_assumptions (varargin);  % Check if assumptions exist - Sympy don't check this
+          end
+          asm = varargin;
         end
-        % ---------------------------------------------
       end
 
-      return
+      isnumber = isnumeric (x) || islogical (x);
+      assert (isempty (asm) || ~isnumber, 'Only symbols can have assumptions.')
 
-    else % S() in other case
+      if (~isscalar (x) && isnumber)  % Handle octave numeric matrix
+        error ('conversion to numeric array of symbols from the constructor is no longer supported. Use sym.symarray instead.');
```

Comment:
I replaced this error with `s = sym.symarray(x);` and it seems fine?  Can you look into this?  Lots of tests changed b/c of this...